supabase/lib/client_options.py

```diff
@@ -42,17 +40,19 @@ def replace(
         headers: Optional[Dict[str, str]] = None,
         auto_refresh_token: Optional[bool] = None,
         persist_session: Optional[bool] = None,
-        detect_session_in_url: Optional[bool] = None,
-        local_storage: Optional[Dict[str, Any]] = None,
+        local_storage: Optional[SyncSupportedStorage] = None,
         realtime: Optional[Dict[str, Any]] = None,
         fetch: Optional[Callable] = None,
     ) -> "ClientOptions":
         """Create a new SupabaseClientOptions with changes"""
-        changes = {
-            key: value
-            for key, value in locals().items()
-            if key != "self" and value is not None
-        }
-        client_options = dataclasses.replace(self, **changes)
-        client_options = copy.deepcopy(client_options)
+        client_options = ClientOptions()
+        client_options.schema = schema or self.schema
+        client_options.headers = headers or self.headers
+        client_options.auto_refresh_token = (
+            auto_refresh_token or self.auto_refresh_token
+        )
+        client_options.persist_session = persist_session or self.persist_session
+        client_options.local_storage = local_storage or self.local_storage
+        client_options.realtime = realtime or self.realtime
+        client_options.fetch = fetch or self.fetch
```

Comment:
It would be nice to have a way to do this as a batch, otherwise this will increase in size for every property we add. I don't exactly know how or if it's possible though, but it's worth investigating a bit, don't you think @leynier?

Reply:
Closing this since it works and we want to release soon. Let's investigate it and later on open a new PR to improve this.